futures-macro-async/src/lib.rs

```diff
@@ -631,4 +639,62 @@ if_nightly! {
             |p| AsyncStreamArgs(p.map(|d| d.1.into_iter().collect()).unwrap_or_default())
         ));
     }
+
+    fn await() -> Tokens {
+        quote_cs! {
+            #[allow(unused_macros)]
+            macro_rules! __await {
+                ($e:expr) => ({
+                    let mut future = ::futures::__rt::pinned($e);
```

Comment:
I would just use `new_unchecked` here, and add a fake borrow across the yield point to make sure we're not in a movable generator.